### PR TITLE
Use Java 11 language features where possible

### PR DESCRIPTION
**Pull request by basil**
Testing done: [Core test suite passed](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6700/13/), [BOM test suite passed](https://ci.jenkins.io/job/Tools/job/bom/job/PR-1234/5/)

### CODE REVIEW

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by timja — APPROVED** [review]

**Pull request merged**
